swabbie-redis/src/test/kotlin/com/netflix/spinnaker/swabbie/redis/RedisNotificationQueueTest.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.swabbie.redis
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.netflix.spinnaker.config.resourceDeserializerModule
+import com.netflix.spinnaker.kork.jedis.EmbeddedRedis
+import com.netflix.spinnaker.kork.jedis.JedisClientDelegate
+import com.netflix.spinnaker.kork.jedis.RedisClientSelector
+import com.netflix.spinnaker.swabbie.notifications.NotificationTask
+import com.netflix.spinnaker.swabbie.test.WorkConfigurationTestHelper
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import redis.clients.jedis.JedisPool
+import strikt.api.expectThat
+import strikt.assertions.isEqualTo
+import strikt.assertions.isFalse
+import strikt.assertions.isTrue
+import strikt.assertions.size
+
```

Comment:
Something that Fletch does that I really like is he sets up a base test class (for example, [ResourceRepositoryTests](https://github.com/spinnaker/keel/blob/master/keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepositoryTests.kt)) and then each repo type (in memory, sql, redis, etc) extends that test to provide whatever initialization logic they need [InMemory](https://github.com/spinnaker/keel/blob/master/keel-core/src/test/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryResourceRepositoryTest.kt), [sql](https://github.com/spinnaker/keel/blob/master/keel-sql/src/test/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepositoryTests.kt)).

This makes it so that you're _sure_ the in memory and real implementations work the same. Something to keep in mind for the future!

Reply:
Got plans to write a proper `tck` at some point.